sonic-thermalctld/scripts/thermalctld

```diff
@@ -610,19 +614,18 @@ class TemperatureUpdater(logger.Logger):
                         self._refresh_temperature_status(sfp_name, thermal, thermal_index)
 
                 for psu_index, psu in enumerate(module.get_all_psus()):
-                    psu_name = '{} PSU {}'.format(module_name, psu_index + 1)
-                    for thermal_index, thermal in enumerate(psu.get_all_thermals()):
-                        if self.task_stopping_event.is_set():
-                            return
-
-                        available_thermals.add((thermal, psu_name, thermal_index))
-                        self._refresh_temperature_status(psu_name, thermal, thermal_index)
-
-
-            thermals_to_remove = self.module_thermals - available_thermals
-            self.module_thermals = available_thermals
-            for thermal, parent_name, thermal_index in thermals_to_remove:
-                self._remove_thermal_from_db(thermal, parent_name, thermal_index)
+                    if psu.get_presence() and psu.get_powergood_status():
```

Comment:
@gregoryboudreau why don't we want to add PSU which is present but does not have good power status to the DB?

Reply:
This is two fold, first is to remove stale sensors from the db in the case where a psu is present and active at first but during runtime is disabled while second is that adding sensors for devices that are not in good power status introduces potentially undefined behavior of introducing sensors to the db that are from a not active or faulting PSU.

Reply:
@gregoryboudreau doesn't the user want to see the status of the sensor if its not in good power status? Why do you want to hide it from the user/CLI output? I think it would be better to add another column in the output for "Status" which will indicate the current status of the sensor: "Present", "Absent", "Error", "Disabled", "Not Supported" etc..